File: scala-examples/hello.scala
import scala.io.StdIn

object Hello {
  def main(args: Array[String]) {
    println("What's your name?")
    StdIn.readLine() match {
      case name: String => println(s"Hello, $name")
      case _ => println("Sorry, I didn't hear your null pointer")
    }
  }
}

File: rust-examples/src/hello.rs
use std::io;

fn main() {
    println!("What's your name?");
    let mut name = String::new();
    match io::stdin().read_line(&mut name) {
        Ok(_) => println!("Hello, {}", name),
        Err(e) => println!("Sorry, {}", e)
    }
}
